File: readability.py
import cs50


def main():

    text = cs50.get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = (letters/words) * 100
    S = (sentences/words) * 100

    grade = round(0.0588 * L - 0.296 * S - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    letter = 0
    for char in text:
        if char.isalpha():
            letter += 1
        else:
            letter += 0
    return letter


def count_words(text):
    word = 0
    for char in text:
        if char == " ":
            word += 1
        else:
            word += 0
    return word + 1


def count_sentences(text):
    sentence = 0
    for char in text:
        if char == "." or char == "?" or char == "!":
            sentence += 1
        else:
            sentence += 0
    return sentence


if __name__ == "__main__":
    main()

File: readability.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int count_letters(string text1);
int count_words(string text2);
int count_sentences(string text3);

int main(void)
{
    string text = get_string("Text: ");

    int letter1 =  count_letters(text);

    int words1 = count_words(text);

    int sentence1 = count_sentences(text);

    float L = (float)letter1 / (float)words1 * 100;

    float S = (float)sentence1 / (float)words1 * 100;

    //formula for calculating the readability of a text

    int index = round((0.0588 * L) - (0.296 * S) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

//function to count number of letters in a string

int count_letters(string text1)
{
    int letters = 0;

    for (int i = 0, n = strlen(text1); i < n; i++)
    {
        if (isalpha(text1[i]))
        {
            letters = letters + 1;
        }
        else
        {
            letters = letters + 0;
        }
    }

    return letters;
}

//function to count number of words in a string

int count_words(string text2)
{
    int words = 0;

    for (int j = 0, l = strlen(text2); j < l; j++)
    {
        if (isspace(text2[j]))
        {
            words = words + 1;
        }
        else
        {
            words = words + 0;
        }
    }

    return words + 1;
}

//function to count number of sentences in a string

int count_sentences(string text3)
{
    int sentences = 0;

    for (int k = 0, m = strlen(text3); k < m; k++)
    {
        if (text3[k] == '.' || text3[k] == '?' || text3[k] == '!')
        {
            sentences = sentences + 1;
        }
        else
        {
            sentences = sentences + 0;
        }
    }

    return sentences;
}